src/Microsoft.ML.Transforms/Text/StopWordsRemovingTransformer.cs

```diff
@@ -438,6 +441,45 @@ protected override Delegate MakeGetter(DataViewRow input, int iinfo, Func<int, b
                 return del;
             }
 
+            public void SaveAsOnnx(OnnxContext ctx)
+            {
+                const int minimumOpSetVersion = 9;
+                ctx.CheckOpSetVersion(minimumOpSetVersion, LoaderSignature);
+
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    var srcVariableName = ctx.GetVariableName(_parent.ColumnPairs[i].inputColumnName);
+                    if (!ctx.ContainsColumn(srcVariableName))
+                        continue;
+                    var dstVariableName = ctx.AddIntermediateVariable(_types[i], _parent.ColumnPairs[i].outputColumnName);
+                    SaveAsOnnxCore(ctx, i, srcVariableName, dstVariableName);
+                }
+            }
+
+            private void SaveAsOnnxCore(OnnxContext ctx, int iinfo, string srcVariableName, string dstVariableName)
+            {
+                var opType = "Squeeze";
+                var squeezeOutput = ctx.AddIntermediateVariable(null, "SqueezeOutput", true);
+                var node = ctx.CreateNode(opType, srcVariableName, squeezeOutput, ctx.GetNodeName(opType), "");
+                node.AddAttribute("axes", new long[] { 0 });
+
+                opType = "StringNormalizer";
+                var stringNormalizerOutput = ctx.AddIntermediateVariable(null, "StringNormalizerOutput", true);
+                node = ctx.CreateNode(opType, squeezeOutput, stringNormalizerOutput, ctx.GetNodeName(opType), "");
```

Comment:
Please try to add explicit types in all calls to AddIntermediateVariable.

Reply:
I know that types don't make it to the model if skip (mapping dataviews) is set to true. Should we still add types in that case for readability?

Reply:
Yes, as far as possible please add types and shapes. 

---
In reply to: [449282071](https://github.com/dotnet/machinelearning/pull/5279#discussion_r449282071) [](ancestors = 449282071)

Reply:
If you add a type, but set skip=true, there is no impact. Do you get an error if you don't set skip=true?

---
In reply to: [449288767](https://github.com/dotnet/machinelearning/pull/5279#discussion_r449288767) [](ancestors = 449288767,449282071)

Reply:
According to the documentation of [StringNormalizer](https://github.com/onnx/onnx/blob/master/docs/Operators.md#StringNormalizer) the output data type is a tensor of strings. So shouldn't the data type be `new VectorDataViewType(TextDataViewType)`?  (You have mentioned that you don't know the size ahead of time. Maybe you can leave it as `size=0`)

---
In reply to: [453111374](https://github.com/dotnet/machinelearning/pull/5279#discussion_r453111374) [](ancestors = 453111374,449288767,449282071)

Reply:
I see. This is what I was talking about this morning. I see now that you mean adding just adding the type info while not skipping the mapping. I think this may work if the dimension remains the same, but since the dimension in this case switches from 2-d to 1-d back to 2-d, OnnxRuntime complains because it's expecting the data to stay 2-d. Even when replacing `_type[iinfo]` with `new VectorDataViewType(TextDataViewType)`, the issue remains.

Reply:
Ok. Let us revisit this after this release.

---
In reply to: [453120111](https://github.com/dotnet/machinelearning/pull/5279#discussion_r453120111) [](ancestors = 453120111)